src/Microsoft.DotNet.Tools.Publish/PublishCommand.cs

```diff
@@ -14,124 +14,74 @@
 
 namespace Microsoft.DotNet.Tools.Publish
 {
-    public class Program
+    public class PublishCommand
     {
-        public static int Main(string[] args)
-        {
-            DebugHelper.HandleDebugSwitch(ref args);
+        public string ProjectPath;
+        public string Configuration;
+        public string OutputPath;
+        public string Framework;
+        public string Runtime;
+        public NuGetFramework NugetFramework;
+        public IEnumerable<ProjectContext> ProjectRidFxPermutations;
```

Comment:
I don't know what FizzBuzz Enterprise Edition is, but this is my opinion on this:

When we give a class any public member we're defining an api between that class and anything that uses it. If that API is a field, the class has no control over how and when that field is modified. If at some point we decide we need to control the allowed values, or provide some custom get logic, it requires a larger code change in the class and potentially breaks it's API, requiring even larger code changes.

With the simple addition of `{ get; set; }` right now, we avoid that situation altogether. At a conceptual level, it provides better encapsulation between classes, so they control their own state despite having a public API.